### PR TITLE
Add Crosvm, minijail, update vulkan_headers, vulkan_icd_loader, virglrenderer, libdrm

### PR DESCRIPTION
**Pull request by satmandu**
- only for `x86_64` (work on getting this to work on arm chromebooks would be welcomed.)
- depends on https://github.com/chromebrew/chromebrew/pull/7280
```
chronos@localhost /usr/local/lib/crew/packages (master *%|SPARSE=)$ crosvm --help
Usage: crosvm [--extended-status] [--log-level <log-level>] [--syslog-tag <TAG>] [--no-syslog] <command> [<args>]

crosvm

Options:
  --extended-status use extended exit status
  --log-level       specify log level, eg "off", "error", "debug,disk=off", etc
  --syslog-tag      when logging to syslog, use the provided tag
  --no-syslog       disable output to syslog
  --help            display usage information

Commands:
  balloon           Set balloon size of the crosvm instance to `SIZE` bytes
  balloon_stats     Prints virtio balloon statistics for a `VM_SOCKET`
  battery           Modify battery
  create_qcow2      Create Qcow2 image given path and size
  device            Start a device process
  disk              Manage attached virtual disk devices
  make_rt           Enables real-time vcpu priority for crosvm instances started
                    with `--delay-rt`
  resume            Resumes the crosvm instance
  run               Start a new crosvm instance
  stop              Stops crosvm instances via their control sockets
  suspend           Suspends the crosvm instance
  powerbtn          Triggers a power button event in the crosvm instance
  sleepbtn          Triggers a sleep button event in the crosvm instance
  gpe               Injects a general-purpose event into the crosvm instance
  usb               Manage attached virtual USB devices.
  version           Show package version.
  vfio              add/remove host vfio pci device into guest
  devices           Start one or several jailed device processes.

chronos@localhost /usr/local/lib/crew/packages (master *%|SPARSE=)$ crosvm version
crosvm 0.1.0
[2022-08-17T15:01:45.794402455+00:00 INFO  crosvm] exiting with success
```

Works properly:
- [x] `x86_64`

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=crosvm CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/crosvm.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  })
+
+  depends_on 'libcap'
+  depends_on 'dtc'
+  depends_on 'protobuf'
+  depends_on 'rust' => :build
+  depends_on 'wayland_protocols' => :build
+
+  def self.build
+    @pwd = `pwd`.chomp
```
Comment (supechicken) [inline packages/crosvm.rb:29]: `@pwd = Dir.pwd`?

Comment (satmandu) [reply packages/crosvm.rb]: done.

File: tools/packages.yaml [hunk 2]
```diff
@@ -989,6 +989,11 @@ url: https://github.com/crossmob/CrossMobile/releases/
 activity: medium
 ---
 kind: url
+name: crosvm
+url: https://chromium.googlesource.com/crosvm/crosvm.
```
Comment (supechicken) [inline tools/packages.yaml:993]: Extra `.`?

**Comment by supechicken** [comment]
I am thinking will it affect `crostini` and `arcvm` (they are in the same `dbus` channel)

File: tools/packages.yaml (hunk 2 above)
Comment (satmandu) [reply tools/packages.yaml]: nice catch!

**Comment by satmandu** [comment]
> I am thinking will it affect `crostini` and `arcvm` (they are in the same `dbus` channel)

The goal is to eventually create a much less locked down crostini, so yes this is intended as an eventual replacement. but teh system crosvm is in `/usr/bin`...

File: packages/crosvm.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    FileUtils.mkdir 'build_bin'
+    FileUtils.ln_s "#{CREW_PREFIX}/bin/gcc", 'build_bin/arm-linux-gnueabihf-gcc' if ARCH == 'armv7l'
+    system "PATH=#{@pwd}/build_bin:$PATH cargo build --release"
+  end
+
+  def self.install
+    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/bin/"
+    FileUtils.install 'target/release/crosvm', "#{CREW_DEST_PREFIX}/bin/crosvm", mode: 0o755
+    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/share/policy/crosvm/"
+    FileUtils.cp_r Dir.glob("seccomp/#{ARCH}/*"), "#{CREW_DEST_PREFIX}/share/policy/crosvm/" if ARCH == 'x86_64'
```
Comment (uberhacker) [inline packages/crosvm.rb:39]: Use `Dir["seccomp/#{ARCH}/*"]` shorthand to simplify?

**Review by uberhacker — CHANGES_REQUESTED** [review]
See comments.

File: packages/crosvm.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  depends_on 'libcap'
+  depends_on 'dtc'
+  depends_on 'protobuf'
+  depends_on 'rust' => :build
+  depends_on 'wayland_protocols' => :build
+
+  def self.build
+    @pwd = Dir.pwd
+    FileUtils.mkdir 'build_bin'
+    FileUtils.ln_s "#{CREW_PREFIX}/bin/gcc", 'build_bin/arm-linux-gnueabihf-gcc' if ARCH == 'armv7l'
```
[the remark this answers is not in the document]
Comment (satmandu) [reply packages/crosvm.rb:31]: I don't have it building on any arm architecture yet.

**Comment by satmandu** [comment]
Adding more updated deps for crosvm...

**Comment by satmandu** [comment]
(Still waiting on i686 builds and yes minijail needs to be added to packages.yaml.)

File: packages/crosvm.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (uberhacker) [reply packages/crosvm.rb:33]: Just sayin'...

[the remark this answers is not in the document]
Comment (satmandu) [reply packages/crosvm.rb]: changed.

**Comment by satmandu** [comment]
Everything that builds has been added.

**Review by uberhacker — APPROVED** [review]

**Pull request merged**